cmd/open.go

```diff
@@ -101,35 +105,41 @@ func runOpen(cfg config.Config, flags *pflag.FlagSet, args []string) error {
 			return err
 		}
 
-		matchingExercises := make([]workspace.Exercise, 0, len(exercises))
+		matchingExerciseMeta := make([]*workspace.ExerciseMetadata, 0, len(exercises))
 		for _, exercise := range exercises {
-			if trackID != "" {
-				if exercise.Track == trackID && exercise.Slug == exerciseID {
-					matchingExercises = append(matchingExercises, exercise)
-				}
-			} else if exercise.Slug == exerciseID {
-				matchingExercises = append(matchingExercises, exercise)
+			metaDir := exercise.MetadataDir()
+			meta, err := workspace.NewExerciseMetadata(metaDir)
+			if err != nil {
+				return err
+			}
+
+			if meta.Exercise != exerciseID {
+				continue
+			}
+
+			if trackID != "" && meta.Track != trackID {
+				continue
+			}
+
+			if meta.Team != teamID {
```

Comment:
Initially, I tried `teamID != "" && meta.Team != teamID`, but the current boolean statement is more in line with how the api handles team exercises.